crates/cheatcodes/spec/src/vm.rs

```diff
@@ -2160,6 +2160,10 @@ interface Vm {
     #[cheatcode(group = Scripting)]
     function signDelegation(address implementation, uint256 privateKey) external returns (SignedDelegation memory signedDelegation);
 
+    /// Sign an EIP-7702 authorization for delegation for specific nonce
+    #[cheatcode(group = Scripting)]
+    function signDelegationWithNonce(address implementation, uint256 privateKey,  uint64 nonce) external returns (SignedDelegation memory signedDelegation);
```

Comment:
I think we can just overload existing  `signDelegation` and `signAndAttachDelegation` fns and accept the nonce param.
Please also run `cargo cheats` in order to generate the json interface as in here: https://github.com/foundry-rs/foundry/blob/master/docs/dev/cheatcodes.md#json-interface

Reply:
The original signAndAttachDelegation is a method that would increase the nonce by default and overwrite in local state. Because the new method signDelegationWithNonce would skip overwrite nonce so I am not so sure. I would fix the CI and something about format and test first.

Reply:
> The original signAndAttachDelegation is a method that would increase the nonce by default and overwrite in local state. Because the new method signDelegationWithNonce would skip overwrite nonce so I am not so sure.

yeah, what I mean is to overload them, something similar with deployCode for example

https://github.com/foundry-rs/foundry/blob/5699fab4c8917d7d0f542fc552dc88afe543ccc7/crates/cheatcodes/spec/src/vm.rs#L1864-L1881